src/nrfx/hal/nrf_clock.c

```diff
@@ -32,14 +32,62 @@ void nrf_clock_task_trigger(NRF_CLOCK_Type * p_reg, nrf_clock_task_t task)
 {
   if (task == NRF_CLOCK_TASK_HFCLKSTART) {
     NRF_CLOCK_regs.TASKS_HFCLKSTART = 1;
+    NRF_CLOCK_regs.HFCLKRUN = 1;
     nrf_clock_reqw_sideeffects_TASKS_HFCLKSTART();
   } else if (task == NRF_CLOCK_TASK_HFCLKSTOP) {
     NRF_CLOCK_regs.TASKS_HFCLKSTOP = 1;
     nrf_clock_reqw_sideeffects_TASKS_HFCLKSTOP();
   } else if (task == NRF_CLOCK_TASK_LFCLKSTART) {
     NRF_CLOCK_regs.TASKS_LFCLKSTART = 1;
+    NRF_CLOCK_regs.LFCLKRUN = 1;
     nrf_clock_reqw_sideeffects_TASKS_LFCLKSTART();
+  } else if (task == NRF_CLOCK_TASK_LFCLKSTOP) {
+    NRF_CLOCK_regs.TASKS_LFCLKSTOP = 1;
+    NRF_CLOCK_regs.LFCLKRUN = 0;
+  } else if (task == NRF_CLOCK_TASK_CAL) {
+    /*We don't want to model clock calibration as per now*/
+    NRF_CLOCK_regs.TASKS_CAL = 1;
+  } else if (task == NRF_CLOCK_TASK_CTSTART) {
+    /*We don't want to model clock calibration as per now*/
+    NRF_CLOCK_regs.TASKS_CTSTART = 1;
+  } else if (task == NRF_CLOCK_TASK_CTSTOP) {
+    /*We don't want to model clock calibration as per now*/
+    NRF_CLOCK_regs.TASKS_CTSTOP = 1;
   } else {
     bs_trace_error_line_time("Not supported task started in nrf_clock\n");
   }
 }
+
+bool nrf_clock_start_task_check(NRF_CLOCK_Type const * p_reg, nrf_clock_domain_t domain)
```

Comment:
Maybe better place this in the header, just like in the real HAL. The hope is to minimize the differences between this replacement HAL and the real Nordic HAL. So for functions which can be the same, better for them to be the same and be placed in the same files as in the Nordic HAL.
(You should only need to add `static inline` before `bool`. And you should not need to do the change of replacing `p_reg->`  with `NRF_CLOCK_regs.`)

Same thing about nrf_clock_lf_srccopy_get and nrf_clock_hf_src_get below.

(In that regard, I should have placed myself `nrf_clock_int_enable_check` in the header...)